pkg/fleet-manager/application/controller.go

```diff
@@ -315,9 +315,21 @@ func (a *ApplicationManager) reconcileSyncStatus(ctx context.Context, app *appli
 	return nil
 }
 
-func (a *ApplicationManager) reconcileDelete(ctx context.Context, app *applicationapi.Application, fleet *fleetapi.Fleet) (ctrl.Result, error) {
-	if err := a.deleteResourcesInMemberClusters(ctx, app, fleet); err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to delete rollout resource in cluster")
+// Handling Application Deletion Based on Fleet Availability
+// When deleting an application, the approach taken depends on whether the managing fleet can be retrieved.
+// If the fleet is available:
+// Application will be removed
+// Resources related to the application will then be deleted from the attachedcluster
+// If the fleet cannot be retrieved:
+// Only the application object itself will be removed
+// Related resources in any cluster will be left intact
+func (a *ApplicationManager) reconcileDelete(ctx context.Context, app *applicationapi.Application) (ctrl.Result, error) {
+	fleetKey := generateFleetKey(app)
+	fleet := &fleetapi.Fleet{}
+	if err := a.Client.Get(ctx, fleetKey, fleet); err == nil {
```

Comment:
we should log error when the error is not found, for other errors we should retry.

Reply:
if the application does not need to delete resources in member cluster, we may skip this ops